app/code/Magento/Shipping/Controller/Adminhtml/Order/ShipmentLoader.php

```diff
@@ -110,7 +110,12 @@ public function load()
         $orderId = $this->getOrderId();
         $shipmentId = $this->getShipmentId();
         if ($shipmentId) {
-            $shipment = $this->shipmentRepository->get($shipmentId);
+            try {
+                $shipment = $this->shipmentRepository->get($shipmentId);
+            } catch (\Exception $e) {
+                $this->messageManager->addErrorMessage(__('This shipment no longer exists.'));
+                return false;
+            }
```

Comment:
You don't need this modification.
If there's `NoSuchEntityException` thrown by `->get()`, then you should let it pass to the method that needs it: ( app/code/Magento/Sales/Controller/Adminhtml/Order/Creditmemo/View.php ) and handle it there.

Reply:
hi @lbajsarowicz thans for reply...
this is for the shipment view not for creditmemo view..and also in view controller there is no try catch added ...so do you want me add try catch in view controller?

Reply:
What I suggest is to remove try { } catch from Loader and put try { } catch with messageManager in controller.